### PR TITLE
Add QtXlsx 1.4.3

### PR DESCRIPTION
**Pull request by MartinDelille**
Specify library name and version:  **qxlsx/1.4.3**

This library handle Qt 6 which qtxlsxwriter do not.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by Jihadist** [comment]
I'm not sure it's a good idea to merge QXlsx without https://github.com/QtExcel/QXlsx/pull/175 because it fully changes CMakeLists. I don't know why this pr was merged and then reverted (with force push).

**Comment by Jihadist** [comment]
Why not to take https://github.com/QtExcel/QXlsx/pull/175 as patch? Because it contains cmake install

**Comment by MartinDelille** [comment]
Yes you're rignt: it'll be easier to insure compatibility if merged.

**Comment by MartinDelille** [comment]
@Jihadist Any idea why windows shared library doesn't contains the symbols ?

**Comment by Jihadist** [comment]
@MartinDelille i'll try it locally

**Comment by Jihadist** [comment]
@MartinDelille try to run it locally, you can find answer here 
```
qxlsx/1.4.3 package(): Packaged 1 '.dll' file: QXlsx.dll
qxlsx/1.4.3 package(): Packaged 19 '.h' files
qxlsx/1.4.3 package(): Packaged 1 file: LICENSE
```
Looks like missing .lib with shared=True :)

**Comment by Jihadist** [comment]
@MartinDelille look here pls https://github.com/MartinDelille/conan-center-index/pull/5

**Review by Jihadist — DISMISSED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

File: recipes/qxlsx/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("qt/5.15.2")
```
Comment (SpaceIm) [inline recipes/qxlsx/all/conanfile.py:43]: if this library handles Qt6 as you said, why not depend on qt6?

File: recipes/qxlsx/all/patches/1.4.3-0001-cmake_allow_shared_build.patch [hunk 2]
```diff
@@ -0,0 +1,198 @@
+diff --git a/QXlsx/CMakeLists.txt b/QXlsx/CMakeLists.txt
```
Comment (SpaceIm) [inline recipes/qxlsx/all/patches/1.4.3-0001-cmake_allow_shared_build.patch:1]:
```suggestion
see https://github.com/QtExcel/QXlsx/pull/185

diff --git a/QXlsx/CMakeLists.txt b/QXlsx/CMakeLists.txt
```

File: recipes/qxlsx/all/test_package/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake", "cmake_find_package_multi"
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self.settings):
```
Comment (SpaceIm) [inline recipes/qxlsx/all/test_package/conanfile.py:15]:
```suggestion
        if not tools.cross_building(self):
```

File: recipes/qxlsx/all/test_package/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.15.6)
+project(test_package CXX)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
```
Comment (SpaceIm) [inline recipes/qxlsx/all/test_package/CMakeLists.txt:5]:
```suggestion
conan_basic_setup(TARGETS)
```

File: recipes/qxlsx/all/test_package/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.15.6)
```
Comment (SpaceIm) [inline recipes/qxlsx/all/test_package/CMakeLists.txt:1]:
```suggestion
cmake_minimum_required(VERSION 3.1)
```

File: recipes/qxlsx/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
```
Comment (SpaceIm) [inline recipes/qxlsx/all/conanfile.py:73]:
```suggestion
        self.cpp_info.libs = ["QXlsx"]
```

File: recipes/qxlsx/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+        self.cpp_info.set_property("cmake_file_name", "QXlsx")
```
Comment (SpaceIm) [inline recipes/qxlsx/all/conanfile.py:74]:
```suggestion
        self.cpp_info.set_property("cmake_file_name", "QXlsx")
        self.cpp_info.set_property("cmake_target_name", "QXlsx::QXlsx")
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]

File: recipes/qxlsx/all/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+        self.cpp_info.names["cmake_find_package"] = "QXlsx"
+        self.cpp_info.names["cmake_find_package_multi"] = "QXlsx"
```
Comment (SpaceIm) [inline recipes/qxlsx/all/conanfile.py:76]: please adapt to your patch which stands that imported target should be `QXlsx::Core`

File: recipes/qxlsx/all/conanfile.py (hunk 1 above)
Comment (Jihadist) [reply recipes/qxlsx/all/conanfile.py:43]: Why it should depend on qt6? KDE still uses qt5, a lot of apps also use qt5.

**Comment by SpaceIm** [comment]
Please consider https://github.com/MartinDelille/conan-center-index/pull/6, it addresses all my reviews (except bump of qt).

File: recipes/qxlsx/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/qxlsx/all/conanfile.py:46]: To have a consistent dependency graph in conan-center.

**Review by SpaceIm — APPROVED** [review]
It's worth noting that it's broken with qt6 (I've tested on github action) because upstream CMakeLists forces C++14, but Qt6 needs at least C++16.

**Review by uilianries — APPROVED** [review]

**Comment by MartinDelille** [comment]
I didn't look at the Qt 6 compatibility yet, but how can I increase C++ requirement?

**Comment by SpaceIm** [comment]
There are 2 issues:
- upstream CMakeLists forces C++14 in its CMakeLists => should be patched to use compile feature `cxx_std_14`
- conan generators are not able to model a C++ standard requirement in CMake targets, so `cxx_std_17` is not transitively injected by qt6 targets => solution: replace `cxx_std_14` on the fly (through tools.replace_in_file) by `cxx_std_17` if major version of qt is 6 or higher.

**Comment by MartinDelille** [comment]
I can't find any reference to cxx_std_14/cxx_std_17 in https://github.com/QtExcel/QXlsx

**Comment by MartinDelille** [comment]
I have a linker error with qt6:

```
...
ld: warning: object file (/Users/martin/.conan/data/libiconv/1.16/_/_/package/e4fbacdee83eb1a2e68227ffe19f6e9a61b36cbc/lib/libiconv.a(relocatable.o)) was built for newer macOS version (11.0) than being linked (10.14)
Undefined symbols for architecture x86_64:
  "_md_parse", referenced from:
      QTextMarkdownImporter::import(QTextDocument*, QString const&) in libQt6Gui.a(qtextmarkdownimporter.cpp.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [bin/test_package] Error 1
make[2]: *** [CMakeFiles/test_package.dir/all] Error 2
make[1]: *** [all] Error 2
ERROR: qxlsx/1.4.3 (test package): Error in build() method, line 12
	cmake.build()
	ConanException: Error 2 while executing cmake --build '/Users/martin/dev/conan/cci/recipes/qxlsx/all/test_package/build/56d08b686b734c6d7e99dc8cb7aca1645123689d' '--' '-j8'
```

**Review by SSE4 — APPROVED** [review]

**Pull request merged**